### ISSUE TITLE
Kundera does not support null in Insert queries for Cassandra

### ISSUE THREAD

**Issue by yeshvanthni** [Impetus/kundera#503]
Null is a valid value in Cassandra 1.2.4 or later and there is no way to update a field to null using kundera. [https://issues.apache.org/jira/browse/CASSANDRA-3783]

Persist and merge for kundera should be handled differently. Merge should fire partial queries with just the non-null values in the insert query like the current behavior. Persist should fire a complete query including the null values.

**Comment by impetus-opensource-admin** [comment]
Support for null is planned with Cassandra 2.x library upgrades. Will see if can be clubbed #478 for 2.10 release

-Vivek

**Comment by dtuzikov** [comment]
Hi,
Is there any progress on this issue? Haven't found fix in CQLTranslator for ver. 2.13.

**Comment by chhavigangwal** [comment]
Marking it for verification for 2.14 release

**Comment by dtuzikov** [comment]
Please check method _appendValue_ in _CQLTranslator_:

``` java
public boolean appendValue(StringBuilder builder, Class fieldClazz, Object value, boolean isPresent,
            boolean useToken) { 
   ... 
   return isPresent;
}
```

It always returns _isPresent_ for null-values, by default it is _false_ (see _CQLTranslator.appendColumnValue_ method). 

May be we can use _externalProperties_ for ability to configure this behaviour.

**Comment by mevivs** [comment]
@dtuzikov 
What if value hasn't been provided, isn't by default null? Also Does Cassandra 2.x allows searches like "isNull" or "not null"?

-Vivek

**Comment by mevivs** [comment]
With entity validation framework in place, i guess we have got null and not null validation place.

-Vivek

**Comment by dtuzikov** [comment]
@mevivs 
As for 2._, Cassandra supports *null_ for insert/update with CQL:

``` sql
INSERT INTO my_table (my_key,my_value) VALUES (1, null);
UPDATE my_table SET my_value = null WHERE my_key=1;
```

Neither _is null_ nor _not null_ supported, setting column to _null_ means deleting it.

If value for column hasn't been provided when inserting new row, it's null by default. But, if i want to set null value for existing not null column and set it for entity _myEntity.setMyValue(null)_, CQLTranslator will ignore it on persist/merge operation and insert query will be without that column. 

``` java
private void onTranslation(TranslationType type, StringBuilder builder, StringBuilder columnBuilder,
            String columnName, Object record, Field column)
    {
        switch (type)
        {
        case ALL:
            if (appendColumnValue(builder, record, column)) // <--- returns false for null-value column
            {
                builder.append(",");
                appendColumnName(columnBuilder, columnName);
                columnBuilder.append(","); // because only key columns
            }
            break;

        case COLUMN:
                 // .....
        case VALUE:
                 // .....
        }
    }
```

So, i should manually call _update my_table set my_value=null where my_row_key='row0'_, it's not convenient way.

Haven't tried validation, i think this is a different task.

**Comment by mevivs** [comment]
@dtuzikov  I see your point!   Perfect and make sense with JPA model

:+1:  to fix it

**Comment by chhavigangwal** [comment]
Fixed with Kundera -2.14 Release
